environments/openshift/prometheus.libsonnet

```diff
@@ -0,0 +1,328 @@
+local k3 = import 'ksonnet/ksonnet.beta.3/k.libsonnet';
+local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
+local list = import 'telemeter/lib/list.libsonnet';
+
+{
+  _config+:: {
+    namespace: 'observatorium',
+
+    versions+:: {
+      prometheusAms: 'v2.12.0',
+      remoteWriteProxy: '14e844d',
+    },
+
+    imageRepos+:: {
+      prometheusAms: 'quay.io/prometheus/prometheus',
+      remoteWriteProxy: 'quay.io/app-sre/observatorium-receive-proxy',
+    },
+
+    ams+:: {
+      proxyPort: 8080,
+      remoteWriteTarget: 'http://%s.%s.svc.cluster.local:%d/api/v1/receive' % [
+        $.thanos.receive.service.metadata.name,
```

Comment:
niceeeee this is using the special thanos-receive service that selects all of the hashrings :)

One question: The thanos-receive service does not have a cluster IP, meaning that when looking up the DNS name `thanos.telemeter-production.svc.cluster.local` you will get back a list of all of the thanos-receive IPs. Do you know how NGINX behaves here with regards to DNS resolution? Does nginx round-robin these IPs? or does it always use the first? etc. 

For this integration, where the amount of traffic is quite low, it's not a big deal, but for integrations with l0ts of data, if NGINX always uses just one IP, then we could take down hashring nodes one by one.

Reply:
OK, looks like we are fine:, per the NGINX docs:

> A domain name that resolves to several IP addresses defines multiple servers at once.
http://nginx.org/en/docs/http/ngx_http_upstream_module.html#server

and

> By default, requests are distributed between the servers using a weighted round-robin balancing method.
http://nginx.org/en/docs/http/ngx_http_upstream_module.html#upstream

So we'll automatically load-balance across all of the IPs